### PR TITLE
JBTM-2579 Throw XAException in XATerminator::commit if a wrapped resource fails transiently

### PR DESCRIPTION
**Pull request by tomjenkinson**
https://issues.jboss.org/browse/JBTM-2579

!BLACKTIE !PERF !XTS !QA_JTS_JDKORB !QA_JTS_JACORB !QA_JTA

### CODE REVIEW

File: ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java [hunk 1]
```diff
@@ -241,6 +241,9 @@ public final int doPhase2Commit ()
         }
 
 		super.phase2Commit(true);
+		if (super.failedList != null && super.failedList.size() > 0) {
+		    return ActionStatus.COMMITTING;
+		}
 
 		/*
 		 * Now do after completion stuff.
```
Comment (mmusgrov) [inline ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java]: A remark: if users have been relying on the old behaviour then their after completions won't run so would it be worthwhile highlighting the fix in the release notes? But I guess this is a corner case you are fixing in which case it probably doesn't need to be specifically brought out?

**Comment by mmusgrov** [comment]
I only reviewed the main changes and did not look at any of the tests. If you do want me to check those too let me know.

File: ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java (hunk 1 above)
Comment (tomjenkinson) [inline ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java]: Sure, I can update the Jira issue with that note.
